### PR TITLE
feat: add callReadOnlyFunction

### PR DESCRIPTION
**Pull request by aulneau**
## Description

Yay, my first contribution to this library! When implementing `makeContractCall` within the explorers sandbox, I ran into some examples of read only calls. I realized this method doesn't apply, took a look at how @hstove had implemented it, and worked on this new method called `callReadOnlyFunction`. 

1. Motivation for change: 
    - We need some ability for folks to call read only functions.
2. What was changed
    - I added the following methods `callReadOnlyFunction`, `cvToHex`. I also fixed some typos and missing jsdoc declarations.
3. How does this impact application developers
    - This method will allow developers who are interacting with contracts to easily call a known read only contract function.

~~4. Link to relevant issues and documentation~~
5. Provide examples of use cases with code samples and applicable acceptance criteria
```tsx
const value = await callReadOnlyFunction({
  senderAddress: identity?.address as string,
  contractAddress,
  contractName,
  functionArgs,
  functionName: func.name,
  network: net, // testnet
});

const parseReadOnlyResponse = ({ result }: ReadOnlyResponse) => {
  const hex = result.slice(2);
  const bufferCv = Buffer.from(hex, 'hex');
  const clarityValue = deserializeCV(bufferCv);
  return cvToString(clarityValue);
};

const result = parseReadOnlyResponse(value);

setResult(result);
```

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
I don't think so.

## Are documentation updates required?
Yes

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [x] Changelog is updated
- [x] Tag 1 of @yknl, @zone117x, @reedrosenbluth for review

### CODE REVIEW

**Comment by friedger** [comment]
Fixed #89

File: src/builders.ts [hunk 1]
```diff
@@ -771,3 +773,82 @@ export function makeContractNonFungiblePostCondition(
...
+/**
+ * Read only function response object
+ *
+ * @param {Boolean} okay - the status of the response
+ * @param {string} result - serialized hex clarity value
+ */
+
+export interface ReadOnlyFunctionResponse {
+  okay: boolean;
+  result: string;
```
Comment (friedger) [inline src/builders.ts:786]: I'd like to see a clarity value here as the result

File: src/builders.ts [hunk 2]
```diff
@@ -771,3 +773,82 @@ export function makeContractNonFungiblePostCondition(
...
+  const {
+    contractName,
+    contractAddress,
+    functionName,
+    functionArgs,
+    network,
+    senderAddress,
+  } = options;
+
+  const url = `${network.coreApiUrl}/v2/contracts/call-read/${contractAddress}/${contractName}/${functionName}`;
```
Comment (friedger) [inline src/builders.ts:836]: function names can contain `?` and should be encoded with `encodeURIComponent`

Comment (aulneau) [reply src/builders.ts:836]: Oh good to know! Do you have a contract that I could test with?

Comment (friedger) [reply src/builders.ts:836]: Something like this?

```
(define-read-only (info?)
  (some u1)
)
```

**Pull request merged**